src/studylist/StudyListWithData.js

```diff
@@ -129,28 +162,72 @@ class StudyListWithData extends Component {
     this.searchForStudies(searchData);
   };
 
+  update = () => {
+    this.setState({
+      modalComponentId: null,
+      showStudyList: true,
+    });
+  };
+
   render() {
     if (this.state.error) {
       return <div>Error: {JSON.stringify(this.state.error)}</div>;
-    } else if (this.state.studies === null) {
+    } else if (this.state.studies === null && !this.state.modalComponentId) {
       return <div>Loading...</div>;
     }
 
+    let healthCare = '';
+    if (this.props.user) {
```

Comment:
Can we put this behind the `window.config.enableGoogleCloudAdapter` variable?

Reply:
done